.goreleaser.yaml

```diff
@@ -0,0 +1,76 @@
+project_name: juno
+before:
+  hooks:
+    - go mod tidy
+    - go generate ./...
+release:
+  draft: true
+  prerelease: auto
+  mode: append
+  header: |
+    ## Some title ({{ .Date }})
+
+    Welcome to this new release!
+
+  footer: |
+    ## Thanks!
+
+    Those were the changes on {{ .Tag }}!
+
+  name_template: "{{.ProjectName}} v{{.Version}}"
+
+builds:
+  - env:
+      - CGO_ENABLED=0
```

Comment:
It just disable usage of CGO. Are you sure we need it? I've successfully built and run Juno without CGO. It appears that since go [v1.20](https://go.dev/doc/go1.20#cgo) CGO is disabled by default(on a system without a C compiler). It reduces dependencies, size and make it more portable. What is more it should even work better in our docker container(I didn't test that yet)

Reply:
If I am not mistaken pebble uses CGO. I wonder how it compiles without it

Reply:
Yeah, indeed Pebble utilizes CGO, but it can also function with CGO disabled. When CGO is disabled, it relies on pure Go implementations for packages in the standard library that would normally use CGO. Based on the benchmarks I've run, Pebble's performance appears to be slightly worse with CGO disabled. E.g:
```
CGO disabled - 7860 blocks took 2h 6 min
root@localhost:/juno/db# CGO_ENABLED=0 go test -bench=. -benchmem -count=100000 -timeout=1h ./...
ok      github.com/NethermindEth/juno/db        16.931s
ok      github.com/NethermindEth/juno/db/pebble 612.908s

CGO enabled - 7860 blocks took 2h 2 min
root@localhost:/juno/db# CGO_ENABLED=1 go test -bench=. -benchmem -count=100000 -timeout=1h ./...
ok      github.com/NethermindEth/juno/db        17.165s
ok      github.com/NethermindEth/juno/db/pebble 401.975s
```

Anyway, it seems that the upcoming Rust VM integration will require CGO. Given this, I will pause my work on this issue to avoid spending more time on something that will likely need to be revisited soon.

Reply:
geth switched to pebble as well, I wonder if they use it with or without CGO.